### PR TITLE
Add IntersectionObserver declarations

### PR DESCRIPTION
**Pull request by pascalduez**
Hi,

add `IntersectionObserver, IntersectionObserverEntry` declarations.

Docs from:
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API  
https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp  
https://developer.mozilla.org/en-US/docs/Web/API/DOMRectReadOnly

### CODE REVIEW

**Comment by billschaller** [comment]
@pascalduez Thanks!

File: lib/bom.js [hunk 1]
```diff
@@ -362,6 +366,55 @@ declare class MutationObserver {
...
+declare type IntersectionObserverOptions = {
+    root?: Node | null,
+    rootMargin?: string,
+    threshold?: number | Array<number>,
+};
+
+declare class IntersectionObserver {
+    constructor(
+      callback: IntersectionObserverCallback,
+      options: IntersectionObserverOptions
```
Comment (julienw) [inline lib/bom.js]: The second argument is actually optional, see https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/IntersectionObserver

File: lib/bom.js [hunk 2]
```diff
@@ -362,6 +366,55 @@ declare class MutationObserver {
...
+    top: number,
+    right: number,
+    bottom: number,
+    left: number,
+    constructor(x: number, y: number, width: number, height: number): void,
+    fromRect(rectangle?: { x?: number, y?: number, width?: number, height?: number }): DOMRect,
+}
+
+declare class DOMRect extends DOMRectReadOnly {
+    constructor(x: number, y: number, width: number, height: number): void,
```
Comment (julienw) [inline lib/bom.js]: These are all optionals too, according to https://drafts.fxtf.org/geometry/#DOMRect

Comment (pascalduez) [inline lib/bom.js]: Thanks @julienw, I will give it another look.

File: lib/bom.js (hunk 1 above)
Comment (Hypnosphi) [reply lib/bom.js]: There is an open PR fixing this: https://github.com/facebook/flow/pull/4664

**Pull request closed without merging**